src/libraries/System.Private.CoreLib/src/System/Collections/HashHelpers.cs

```diff
@@ -15,6 +15,9 @@ internal static partial class HashHelpers
 
         public const int HashPrime = 101;
 
+        // must never be written to
```

Comment:
Nit: seems hard to guarantee this invariant given that it is public and the instance is effectively owned by instances of a different class. I would probably just make this a private field in Dictionary.

Reply:
I put it in here so that it can be shared between `Dictionary` and `HashSet` and so that it doesn't get a new instantiation for every generic type argument combination.